### ISSUE TITLE
[3.3.5/Game objects] Ships in instances

### ISSUE THREAD

**Issue by Freeman256** [TrinityCore/TrinityCore#26770]
### Description

Some time ago, I saw messages from popular private servers that players in instances saw ships, and so I myself saw something similar. While I was making my way into the The Slave Pens in Zangarmarsh, an elven ship appeared at the beginning of the instance.
![WoWScrnShot_080121_141908](https://user-images.githubusercontent.com/73786888/127769093-c36342b2-9ca1-4a99-a4cf-966110db4e8c.jpg)

According to rumors, this happened in the Tempest Keep: The Eye, and there the ships blocked the exit from the instance, but I have not encountered this.

### Expected behaviour

Ships should not appear in instances.

### Steps to reproduce the problem

It's hard to say what this problem is connected with, but I followed this plan:
1. I flew across Hellfire Peninsula as a level 66 alliance character;
2. I went to Zangarmarsh;
3. In Cenarion Refuge, I took quest [Failed Incursion](https://tbc.wowhead.com/quest=9876/failed-incursion) and went to do it;
4. I completed quest [Failed Incursion](https://tbc.wowhead.com/quest=9876/failed-incursion) on enter the Coilfang Reservoir and took quest [Lost in Action](https://tbc.wowhead.com/quest=9738/lost-in-action) from [Watcher Jhang](https://tbc.wowhead.com/npc=17884/watcher-jhang);
5. Then I went to the The Slave Pens and spent 15-20 minutes there;
6. Then I saw that I was in a state of battle and decided to return to the entrance to the instance. There I saw one crab, which was aggroing at me, but at the same time ran along its route and did not pay attention to me, and I also saw a ship.

### Branch

3.3.5

### TC rev. hash/commit

https://github.com/TrinityCore/TrinityCore/commit/caf68840b9bc6a435a59b2fb954c1fe779a27c66
TrinityCore rev. 2021-07-24 ‏‎12:46:23 +0300 (3.3.5 branch) (Win64, Release, Static)

### Operating system

Win7 SP1 x64 with all updates (build 6.1.7601).

### Custom changes

None.

**Comment by Freeman256** [comment]
Here is a screenshot of the same location and with the same angle, in which the ship is not visible. I left the game for a while.
![WoWScrnShot_080121_154207](https://user-images.githubusercontent.com/73786888/127771302-65328fb6-2d78-44dd-8f33-5ed59034b20a.jpg)

**Comment by offl** [comment]
```txt
<Jinnai> very rare
<Jinnai> but can happen
<Jinnai> also with zeppelins in deeprun tram (had that one myself)
<Jinnai> no idea wtf is causing this
<Aokromes> very very very rare
<Aokromes> a lot of years i don't have seen one
<Aokromes> and it's usualy ships spawning on 0, 0, 0
<Aokromes> i remember on kalimdor to see one long time ago
<Aokromes> on EK you cannot see it because 0,0,0 0 is under floor
<Aokromes> https://github.com/TrinityCore/TrinityCore/issues/5235
<Offle> "My friends thought I was having hallucinations as I told them that a Zeppelin was blocking my way in the Deeprun Tram. They didn't see it and could walk right through it but for me it was a solid object."
<Offle> lol
<%Shauren> im sure the transport isnt actually there
<%Shauren> whats more likely is a missed/wrong smsg_update_object with outofrange or smsg_destroy_object
<%TCRelay> <M​odoX> so objects actually persist w/o destroying them in updateobject even when transferring to a different map? thats interesting :D
<%Shauren> well yeah

**Comment by Conscript01** [comment]
my friend posted me this screenshot like 2 days ago. Its very rare and I saw it like 3 times only. It was always in EK/Kalimdor instances


![ship](https://user-images.githubusercontent.com/55281276/129444392-01b2c7d9-471d-42bc-9086-538776835cf0.png)

**Comment by Aokromes** [comment]
i don't have seen reports of this bug for long long long time.

**Comment by Freeman256** [comment]
Indeed, a lot of time has passed... Since then, for some reason, it is impossible to even launch a server on the still best operating system - Windows 7! So I can't even update the server. Maybe we should stop supporting all operating systems up to Windows 13 in advance?! Let's take the situation to the point of absurdity. Refusing to support Windows 7 did not rid the server of all bugs. In general, how many bugs were cured by refusing to support Windows 7? 1? 2? Or none at all? It was stupid and offensive.

**Comment by Aokromes** [comment]
> Indeed, a lot of time has passed... Since then, for some reason, it is impossible to even launch a server on the still best operating system - Windows 7! So I can't even update the server. Maybe we should stop supporting all operating systems up to Windows 13 in advance?! Let's take the situation to the point of absurdity. Refusing to support Windows 7 did not rid the server of all bugs. In general, how many bugs were cured by refusing to support Windows 7? 1? 2? Or none at all? It was stupid and offensive.

blame microsoft who is who doesn't support windows 7 with latest visual studio versions, if you cannot / don't want to run windows 10 or 11 switch to linux to have updated compilers.

**Comment by Freeman256** [comment]
I have 3 computers at home, and they all have Windows 7. I have no desire to reinstall (or install a second) operating system on them because of ONE game. I don't need any updates from Microsoft for the operating system. Compilers don't need to be updated either. It's just a paradoxical situation: the game client 3.3.5a even supports Windows XP, while the server has recently required almost the newest operating system. Some modern modified engines of the DOOM game somehow even support DoS, but here it is impossible to even implement support for Windows 7...

**Comment by Shauren** [comment]
> I have 3 computers at home, and they all have Windows 7. I have no desire to reinstall (or install a second) operating system on them because of ONE game. I don't need any updates from Microsoft for the operating system. Compilers don't need to be updated either. It's just a paradoxical situation: the game client 3.3.5a even supports Windows XP, while the server has recently required almost the newest operating system. Some modern modified engines of the DOOM game somehow even support DOS, but here it is impossible to even implement support for Windows 7...

Why should we support software that has long been abandoned by their authors? (outside of the focus of this project, which is the wow client)

**Comment by Freeman256** [comment]
Really! Why support the server part of a 13-year-old game?! The source code of the 3.3.5a server must be burned immediately! Only the code of the most current add-on should remain! Oh no, it's out of date by a couple of minutes! Burn it too! But seriously, obsolescence doesn't happen because someone said so, but in your head. It's a pity that I don't know how to program, and if I did, I would set myself the task of making the 3.3.5a server 100% compatible with the systems that the client is compatible with at all costs. This version of the game will never become obsolete. Yes, perhaps Windows XP really is too old and unstable (although this is also a subjective opinion), and Vista is just a beta version of Windows 7, but at least it should be compatible with Windows 7, I think, since this OS will also never become obsolete (at least until a modern OS appears that will support all the functionality of Windows 7 (for running old games)). By the way, yes, many of my favorite games simply refuse to run on modern OS! And should I stop liking these games because of this?! You won't get it! I'm more likely to dislike modern OSes. Why does client 3.3.5a work perfectly on modern OSes that weren't even planned in 2012? Is it really that difficult to write code that won't use the exclusive functionality of modern OSes? Compilers ask what the target OS will be. It seemed to me that compatibility with Windows 7 disappeared around the time "get system time" appeared in the code.

**Comment by Ovahlord** [comment]
> Really! Why support the server part of a 13-year-old game?! The source code of the 3.3.5a server must be burned immediately! Only the code of the most current add-on should remain! Oh no, it's out of date by a couple of minutes! Burn it too! But seriously, obsolescence doesn't happen because someone said so, but in your head. It's a pity that I don't know how to program, and if I did, I would set myself the task of making the 3.3.5a server 100% compatible with the systems that the client is compatible with at all costs. This version of the game will never become obsolete. Yes, perhaps Windows XP really is too old and unstable (although this is also a subjective opinion), and Vista is just a beta version of Windows 7, but at least it should be compatible with Windows 7, I think, since this OS will also never become obsolete (at least until a modern OS appears that will support all the functionality of Windows 7 (for running old games)). By the way, yes, many of my favorite games simply refuse to run on modern OS! And should I stop liking these games because of this?! You won't get it! I'm more likely to dislike modern OSes. Why does client 3.3.5a work perfectly on modern OSes that weren't even planned in 2012? Is it really that difficult to write code that won't use the exclusive functionality of modern OSes? Compilers ask what the target OS will be. It seemed to me that compatibility with Windows 7 disappeared around the time "get system time" appeared in the code.

Please get professional help or visit a creative workshop. You need it.
